src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java

```diff
@@ -359,38 +370,78 @@ public void onOpened()
         createXpBar();
         createSkillContent();
 
-        resourceList.setDataProvider(() -> getOpenRequestsOfCitizen().size(), (index, rowPane) ->
-        {
-            final ImmutableList<IRequest> openRequests = getOpenRequestsOfCitizen();
-            if (index < 0 || index >= openRequests.size())
-            {
-                return;
-            }
+        resourceList.setDataProvider(new ScrollingList.DataProvider() {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce this anonymous class number of lines from 64 to at most 40, or make it a named class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1188)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Move this left curly brace to the beginning of next line of code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3ALeftCurlyBraceStartLineCheck)